addons/stock/models/stock_move.py

```diff
@@ -1602,3 +1602,36 @@ def _adjust_procure_method(self):
                 mtso_free_qties_by_loc[move.location_id][move.product_id.id] -= needed_qty
             else:
                 move.procure_method = 'make_to_order'
+
+    def _decrease_initial_demand(self, qty, stream='UP'):
+        """Decrease the initial demand and propage it through a moves chain.
+
+        The propagation is stopped by the end of the chain or either by a done
+        or cancel move. In case the one of the current move in `self` is linked
+        to a business document that cannot be updated, the method return them.
+
+        :param qty: quantity to decrease on the move and the previous ones in the chain
+        :type qty: float, required
+        :return: list of non updatable moves: move and corresponding business that haven't be updated
+        :rtype: list of tuples
+        """
+        assert stream in ('UP', 'DOWN'), 'Unknown stream'
+        done_move_to_return = []
+        for move in self:
+            if move.state in ('done', 'cancel'):
+                if not move.origin_returned_move_id:
+                    done_move_to_return.append((move, move.picking_id))
+                continue
+            move.product_uom_qty -= qty
+            if stream == 'UP':
+                done_move_to_return += move.move_orig_ids._decrease_initial_demand(qty, stream=stream)
+            else:
+                done_move_to_return += move.move_dest_ids.filtered(
+                    lambda m: m.group_id == move.group_id
+                )._decrease_initial_demand(qty, stream=stream)
```

Comment:
why a check o the group_id on the else and not the if ?

Reply:
in case of mto-manufacture, the move (delivery) doesn't have the same group than the move_orig_id one (production)